pkg/util/kubernetes/kubelet/kubelet.go

```diff
@@ -13,66 +13,70 @@ import (
 	"fmt"
 	"io/ioutil"
 	"net/http"
-	"strings"
+	"net/url"
 	"time"
 
+	"github.com/DataDog/datadog-agent/pkg/util/kubernetes"
 	log "github.com/cihub/seelog"
 
 	"github.com/DataDog/datadog-agent/pkg/config"
 	"github.com/DataDog/datadog-agent/pkg/util/docker"
-	"github.com/DataDog/datadog-agent/pkg/util/kubernetes"
 	"github.com/DataDog/datadog-agent/pkg/util/retry"
 )
 
-// Kubelet constants
-const (
-	KubeletHealthPath = "/healthz"
-)
-
 var globalKubeUtil *KubeUtil
 
 // KubeUtil is a struct to hold the kubelet api url
 // Instantiate with GetKubeUtil
 type KubeUtil struct {
-	retry.Retrier
-	kubeletAPIURL string
+	// used to setup the KubeUtil
+	initRetry retry.Retrier
+
+	kubeletApiEndpoint       string
+	kubeletApiClient         *http.Client
+	kubeletApiRequestHeaders *http.Header
+}
+
+// ResetGlobalKubeUtil is a helper to remove the current KubeUtil global
+// It should be called essentially in the tests
```

Comment:
`It is ONLY to be used for tests`

As user classes keep a pointer to the ku, that'll be a nice split brain situation if one class tries to do that